src/main/kotlin/org/jitsi/nlj/util/ClockUtils.kt

```diff
@@ -29,6 +29,47 @@ fun Instant.formatMilli(): String = TimeUtils.formatTimeAsFullMillis(this.epochS
 
 fun Duration.formatMilli(): String = TimeUtils.formatTimeAsFullMillis(this.seconds, this.nano)
 
+/**
+ * Converts this instant to the number of microseconds from the epoch
+ * of 1970-01-01T00:00:00Z.
+ *
+ * If this instant represents a point on the time-line too far in the future
+ * or past to fit in a [Long] microseconds, then an exception is thrown.
+ *
+ * If this instant has greater than microsecond precision, then the conversion
+ * will drop any excess precision information as though the amount in nanoseconds
+ * was subject to integer division by one thousand.
+ *
+ * @return the number of microseconds since the epoch of 1970-01-01T00:00:00Z
+ * @throws ArithmeticException if numeric overflow occurs
+ */
+fun Instant.toEpochMicro(): Long {
+    return if (this.epochSecond < 0 && this.nano > 0) {
+        val micros = Math.multiplyExact(this.epochSecond + 1, 1000_000)
+        val adjustment: Long = (this.nano / 1000 - 1000_000).toLong()
+        Math.addExact(micros, adjustment)
+    } else {
+        val micros = Math.multiplyExact(this.epochSecond, 1000_000)
+        Math.addExact(micros, (this.nano / 1000).toLong())
+    }
+}
+
+/**
+ * Obtains an instance of [Instant] using microseconds from the
+ * epoch of 1970-01-01T00:00:00Z.
+ * <p>
+ * The seconds and nanoseconds are extracted from the specified milliseconds.
+ *
+ * @param epochMicro the number of microseconds from 1970-01-01T00:00:00Z
+ * @return an instant, not null
+ * @throws DateTimeException if the instant exceeds the maximum or minimum instant
+ */
+fun instantOfEpochMicro(epochMicro: Long): Instant {
+    val secs = Math.floorDiv(epochMicro, 1000_000).toLong()
```

Comment:
In my env `toLong()` in not necessary here, but the code doesn't compile because `micros` is an `Int` and `ofEpochSeconds` expects a `Long`. Strange how the GH action succeed.

Reply:
Fixed both.